### PR TITLE
Typedoc documentation

### PR DESCRIPTION
**Pull request by 0xtristan**
- Added proper TS docs
Looking at other repos (solana/web3.js and raydium) they have nicely formatted SDK documentation. Turns out this is all simple and handled by a nice package called Typedoc. Adding this to our docs so they are a bit more professional. Also includes github action to publish to Github pages.

- Updated license in package.json
- Changed package manager from npm to yarn

### CODE REVIEW

**Comment by 0xtristan** [comment]
IMO we should also add a readme on how to contribute etc.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  push:
+    branches: 
+    - main
+    - release/*
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+#   publish_npm:
```
Comment (xjcaa) [inline .github/workflows/deploy.yml:13]: delete?

**Comment by xjcaa** [comment]
> IMO we should also add a readme on how to contribute etc.

Just add a PR?

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (0xtristan) [reply .github/workflows/deploy.yml:13]: Yeah I was leaving in in case we decide to automate it later

**Pull request merged**